pkg/crc/services/dns/dns.go

```diff
@@ -49,15 +49,14 @@ func setupDnsmasq(serviceConfig services.ServicePostStartConfig) error {
 	}
 
 	// Remove the dnsmasq container if it exists during the VM stop cycle
-	_, _, _ = serviceConfig.SSHRunner.Run("sudo podman rm -f crc-dnsmasq")
-
-	// Start the dnsmasq container
-	dnsServerRunCmd := fmt.Sprintf("sudo podman run  --ip %s --name crc-dnsmasq -v %s:/etc/dnsmasq.conf -p 53:%d/udp --privileged -d %s",
-		dnsContainerIP, dnsConfigFilePathInInstance, dnsServicePort, dnsContainerImage)
-	if _, _, err := serviceConfig.SSHRunner.Run(dnsServerRunCmd); err != nil {
-		return err
+	sd := systemd.NewInstanceSystemdCommander(serviceConfig.SSHRunner)
+	if state, err := sd.Status("crc-dnsmasq"); err != nil || state != states.Running {
+		// Create the dnsmasq container
+		if err := sd.Enable("crc-dnsmasq.service"); err != nil {
+			return err
+		}
 	}
-	return nil
+	return sd.Start("crc-dnsmasq.service")
```

Comment:
Looks like this no longer needs the external VM IP, so snc could also take care of `systemd enable`? `crc` would only need to make sure the service is actually running to let the user know something is wrong if it's not running.

Reply:
@cfergeau if we enable it that means it is going to run for all the platform bundle which we don't want for windows/mac where this is taken care by usermode networking.

Reply:
Probably worth mentioning in the commit log that we don't want it always enabled because it's unneeded for user mode networking. Is it harmful to have it running when usermode networking is used? Because it's missing a `sd.Stop("crc-dnsmasq")` + `sd.Disable("crc-dnsmasq.service")` when switching from system networking to user networking.

Reply:
(maybe `crc-dnsmasq.service` and `gvisor-tap-vsock.service` can be made mutually exclusive at the systemd level, but that's just a long term cleanup/improvement, not something required right now).

Reply:
> Because it's missing a sd.Stop("crc-dnsmasq") + sd.Disable("crc-dnsmasq.service") when switching from system networking to user networking.

@cfergeau when you  run it using user mode networking then this block not even run so we don't need to stop/disable the service.

Reply:
We would need to stop/disable if we were reusing the same VM. But I forgot that it's required to recreate the VM when switching from one mode to the other, so that indeed takes care of it.